pyperformance/data-files/benchmarks/bm_generators/run_benchmark.py

```diff
@@ -0,0 +1,48 @@
+"""
+Benchmark recursive generators implemented in python
+by traversing a binary tree.
+
+Author: Kumar Aditya
+"""
+
+from __future__ import annotations
+
+from collections.abc import Iterator
+
+import pyperf
+
+
+class Tree:
+    def __init__(self, value: int, left: Tree | None, right: Tree | None) -> None:
```

Comment:
FWIW maybe `value` should be the second parameter, since the iterator iterates in in-order? The values would then be provided in the same order in which they are produced.

Reply:
Done